src/firefly/js/ui/DateTimePickerField.jsx

```diff
@@ -80,94 +81,76 @@ function isSameTimes(time1, time2) {
      return compareMoments(m1, m2);
 }
 
-export class DateTimePicker extends PureComponent {
-    constructor(props) {
-        super(props);
-        this.onClose = this.onClose.bind(this);
-        this.onSelectedDate = this.onSelectedDate.bind(this);
+export function DateTimePicker({showInput,openPicker, value='', onChange, onChangeOpenStatus,wrapperStyle,
+                                   inputStyle, groupKey, timeFieldKey}) {
 
-        const {value = ''} = props;
-        const timeMoment = aMoment(value).isValid() ? aMoment(value) : value;   // a moment or original string value
-        this.state={timeMoment};  // store valid moment or string
-    }
-
-    onClose() {
-        const {onChangeOpenStatus} = this.props;
+    const [timeMoment, setTimeMoment] = useState(aMoment(value).isValid() ? aMoment(value) : value);
+    useEffect(() => {
+        //componentDidMount code
+        let iAmMounted = true;
+        let unbinder;
+        // update the state in case start time or end time are updated due to the change from the entry
+        if (groupKey && timeFieldKey) {
+           unbinder = FieldGroupUtils.bindToStore(groupKey, (fields) => {
+                const time = get(fields, [timeFieldKey, 'value']);
+                if (time) {
+                    if (!isSameTimes(time, timeMoment)) {
+                        setTimeMoment(time);
+                    }
+                }
+            });
+        }
 
+        //componentWillUnmount code
+        return () => {
+            if (unbinder) unbinder();
+            iAmMounted = false;
+        };
+    }, []);
```

Comment:
I told you I would suggest another way to do this in the code review. Here it is. using our recently written `useFieldGroupValue` (it is kind of like `useState`)

```javascript
   const [setTimeStr,getTimeStr]= useFieldGroupValue(timeFieldKey);

   useEffect(()  => {
         const time= getTimeStr();
         if (!time || !isSameTimes(time, timeMoment)) return;
        setTimeMoment(time);
  }, [getTimeStr]);
```

If you are having trouble make this work. Then we can look at it together. The hook is pretty new so there might be some issues.

Reply:
Hi Trey, should the if condition above be this instead (removed the 'not' from isSameTimes): 
 
```javascript
         if (!time || isSameTimes(time, timeMoment)) return;
```

to retain the logic of the original code? We only call setTimeMoment(time) when time != timeMoment, correct?

Reply:
opps. good catch. yes it should have been
```
if (!time || !isSameTimes(time, timeMoment)) return;
```